### ISSUE TITLE
Playlist "share" link doesn't get parsed properly

### ISSUE THREAD

**Issue by Crocross** [Androz2091/discord-player#1808]
**Describe the bug**
If you grab the share link of a YT playlist it doesn't seem to be able to resolve it (without manually removing the share portion).

**To Reproduce**
1. Go to a YT playlist page (ex. https://www.youtube.com/playlist?list=PLRxX1Jhp-oqUhk_VQPuyVxwVhRPeuxNYQ)
2. Click on the share icon and copy the link (ex. https://youtube.com/playlist?list=PLRxX1Jhp-oqUhk_VQPuyVxwVhRPeuxNYQ&si=9vQkdM4MnJl_H6HZ)
3. Try to use that link in a bot 
4. If you remove the &si part if towrks (ex. https://youtube.com/playlist?list=PLRxX1Jhp-oqUhk_VQPuyVxwVhRPeuxNYQ)

**Expected behavior**
The link is parsed and resolves correctly.

**Please complete the following information:**
 - Node Version: 19.6.0
 - Discord Player Version: 6.6.3
 - Discord.js Version: 14.13.0

**Additional context**
I tested a tracks share link and that works with no issue although the link format is different as well for tracks: https://youtu.be/BnSkt6V3qF0?si=eecB9Cbpaa_rBT-E

**Comment by Crocross** [comment]
Before I get into the options I see possible I'll differentiate the type of URL patterns I noticed.

- The link used is the [share link of a playlist](https://youtube.com/playlist?list=PLRxX1Jhp-oqUhk_VQPuyVxwVhRPeuxNYQ&si=mscUDYbeja2HCfM0) which is what the issue was originally about. 
- The link used is the [URL when on a track in a playlist](https://www.youtube.com/watch?v=BnSkt6V3qF0&list=PLRxX1Jhp-oqUhk_VQPuyVxwVhRPeuxNYQ&index=1)

The difference in the URL pattern is whether or not the `list=` is the first URL parameter or not.

I think for the case of the first URL type it makes sense to always return a playlist.
For the second URL type there are a few ways to handle it:

1. Return the playlist the track belongs to
2. Return the track only
3. Return the playlist followed by the track(s) on the playlist

I don't know how easy/difficult it is to do the third option but I added it anyhow. The other changes would be minor and shouldn't be very difficult.
